core/lib/dal/src/tee_proof_generation_dal.rs

```diff
@@ -65,58 +63,68 @@ impl TeeProofGenerationDal<'_, '_> {
                         SKIP LOCKED
                 )
             RETURNING
-                tee_proof_generation_details.l1_batch_number
+                tee_proofs.l1_batch_number
             "#,
+            TeeProofGenerationJobStatus::PickedByProver as TeeProofGenerationJobStatus,
```

Comment:
Are these conversions needed? This looks like no-op.

Reply:
I thought the same until I tried removing `as TeeProofGenerationJobStatus`:
```
   Compiling zksync_dal v0.1.0 (zksync-era/core/lib/dal)
error: unsupported type tee_proofs_job_status for param #1
  --> core/lib/dal/src/tee_proof_generation_dal.rs:35:21
   |
35 |           let query = sqlx::query!(
   |  _____________________^
36 | |             r#"
37 | |             UPDATE tee_proofs
38 | |             SET
...  |
74 | |             &processing_timeout,
75 | |         );
   | |_________^
   |
   = note: this error originates in the macro `$crate::sqlx_macros::expand_query` which comes from the expansion of the macro `sqlx::query` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `zksync_dal` (lib test) due to 1 previous error
error: `cargo check` failed with status: exit status: 101
   Compiling zksync_dal v0.1.0 (zksync-era/core/lib/dal)
error: unsupported type tee_proofs_job_status for param #1
  --> core/lib/dal/src/tee_proof_generation_dal.rs:35:21
   |
35 |           let query = sqlx::query!(
   |  _____________________^
36 | |             r#"
37 | |             UPDATE tee_proofs
38 | |             SET
...  |
74 | |             &processing_timeout,
75 | |         );
   | |_________^
   |
   = note: this error originates in the macro `$crate::sqlx_macros::expand_query` which comes from the expansion of the macro `sqlx::query` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `zksync_dal` (lib test) due to 1 previous error
error: `cargo check` failed with status: exit status: 101
Error: Child process exited with code 1
```
Apparently, it's needed for some reason. 🤷🏼‍♂️

Also, please note that we already have similar code merged into the `main` branch.
https://github.com/matter-labs/zksync-era/blob/b45aa9168dd66d07ca61c8bb4c01f73dda822040/core/lib/dal/src/tee_verifier_input_producer_dal.rs#L112-L114